new/cookbook_metadata.md

```diff
@@ -0,0 +1,63 @@
+---
+RFC: unassigned
+Title: Cookbook Metadata Files
+Author: Charles Johnson <charles@chef.io>
+Author: Thom May <tmay@chef.io>
+Status: Draft
+Type: <Standards Track>
+---
+
+# Cookbook Metadata Files
+
+The purpose of this RFC is to clarify the handling of cookbook metadata files by tooling in the Chef ecosystem.
+
+## Motivation
+
+    As a person who writes and shares cookbooks,
+    I want to specify metadata without hand-editing json,
+    because json is a serializable data format never intended for human interaction.
+
+    As a person who uploads cookbooks to a Chef Supermarket server,
+    I want to be able to upload my files without worrying about the format in which the cookbook metadata file was written,
+    because this detail provides me with no tangible value.
+
+    As a person who utilizes cookbooks downloaded from a Chef Supermarket in my own recipes,
+    I should be able to upload those cookbooks to my own Chef Server without worrying about the format in which the cookbook metadata file was written,
+    because this detail provides me with no tangible value.
+
+    As a person who runs a Supermarket server,
+    I should be able to accept cookbook uploads without my server interpreting code as part of the upload process,
+    so that I can more easily provide a secure cookbook sharing service.
+
+## Guiding UX Principles
+1. Machine interfaces should prefer `json` to `rb`
+2. Human interfaces must only require `rb` format metadata files.
+3. Humans must never be required to edit `json` files.
+4. Humans must never be required to care about the format of their cookbook metadata file.
+
+## Specification
+
+- Newly generated cookbooks must be created with only a metadata.rb file.
+- Newly generated cookbooks must chefignore the metadata.rb file, and gitignore the metadata.json file.
+- All server-oriented tools (Supermarket, Chef server, server-side depsolvers, et al.) must only care about the metadata.json, file, and must ignore the metadata.rb if present. If no metadata.json file is present, these tools must fail rather than trying to interpret the metadata.rb to generate a metadata.json.
+- Chef-client must only interpret metadata.json files. If a metadata.rb file is present, the chef-client may use the metadata.rb file to generate a metadata.json file at runtime.
+- Tooling that takes cookbooks from machines and makes them available to humans for editing must locally generate a metadata.rb from the metadata.json, overwiting any metadata.rb that may already exist in the cookbook.
```

Comment:
this rule is caused by the original mistake of having tooling that strips the metadata.rb file out, and it feels like we're doubling down on that confusion.

we need a clearer rule here that metadata.json must always be favored, and metadata.rb must never be stripped (at least not by default) and that metadata.rb should be included in tarballs shipped to the supermarket.

the end users chefignore settings should really determine if they're pushing metadata.rb to their chef server or not (if they want an artifact repo were they can download cookbooks, mutate them and then re-upload then they'd want metadata.rb files uploaded, most other users would want them stripped by default).

stripping metadata.rb is an opinion and tooling should really be more neutral since not everyone shares the same opinion.